### ISSUE TITLE
Setting gateway on an interface has no effect, but HTTP code 200 despite!

### ISSUE THREAD

**Issue by mitch40** [jaredhendrickson13/pfsense-api#229]
I would like to report an unintended behavior (**pfsense 2.6.0 + API 1.4.1**), when I try to set an existing gateway to an already configured interface.

To do this, I perform a **PUT** on _/api/v1/interface,_

Data sent via request:
`{'client-id': 'admin', 'client-token': 'pfsense', 'id': 'em1', 'gateway': 'TRUEWAN', 'apply': True}`

Data received in return:
`{'status': 'ok', 'code': 200, 'return': 0, 'message': 'Success', 'data': {'if': 'em1', 'enable': '', ' spoofmac': '', 'descr': 'INETBOX', 'ipaddr': '192.168.1.164', 'subnet': '24'}}`

Code **200**... so the request seems valid.

When I check with a web browser, the gateway has not been applied on the interface in question. However, the gateway does appear in the "**IPv4 Upstream gateway**" drop-down list in the "**Static IPv4 Configuration**" section.

Is this an error/misuse on my part, or is it a bug?

Thank you for your analysis
![screen](https://user-images.githubusercontent.com/48313048/168481206-addfde73-40ab-4f64-9c9a-b3ca4fef20d0.png)
.

**Comment by jaredhendrickson13** [comment]
Hey!

TL;DR: This isn't a bug, but it looks like a lapse in the documentation. Add `"type": "staticv4"` to your request body and it should work. 

In more detail, the /api/v1/interface endpoint does not support fully partial updates in some situations due to the amount of conditional validation that is required. In this case, the `gateway` field is dependent on `type` being set to `staticv4`. The `type` is not stored in the configuration, at least not directly; so the API expects the client to specify the `type` in order to update or set those fields rather than inferring the type. Currently, the documentation does not list the `type` as a required field, which is technically correct. However, it should detail somewhere that it _may be_ required to update dependent fields like `gateway`. That is something I can plan to include in the next patch.

Let me know if you have any further trouble. 

Thanks!

**Comment by mitch40** [comment]
Thank you for your quick answer, i've tried  with the `staticv4`, it works :)
